python/pyspark/sql/column.py

```diff
@@ -656,9 +668,9 @@ def __iter__(self) -> None:
     """
 
     contains = _bin_op("contains", _contains_doc)
-    rlike = _bin_op("rlike", _rlike_doc)
-    like = _bin_op("like", _like_doc)
-    ilike = _bin_op("ilike", _ilike_doc)
+    rlike = _bin_op("rlike", _rlike_doc, is_other_str=True)
```

Comment:
Or @xinrong-databricks we can alternatively just define them as a proper function.

Reply:
Sounds good! Defined `_bin_op_other_str`.